src/dmd/dmodule.d

```diff
@@ -1237,11 +1236,8 @@ extern (C++) final class Module : Package
         {
             // Add all symbols into module's symbol table
             symtab = new DsymbolTable();
-            for (size_t i = 0; i < members.dim; i++)
-            {
-                Dsymbol s = (*members)[i];
+            foreach (Dsymbol s; *members)
```

Comment:
Do not replace looks over members with foreach. The reason is because many routines, as part of their work, add members to `members` and foreach assumes that the loop length does not change.

Reply:
Thanks, I reverted those changes. I noticed that the for-loop over members is a very common pattern in dmd, so I don't want to make changes to it only in dmodule. I might try to add an `InputRange` to iterate over Dsymbol members in a future PR, because currently it's not at all clear whether these for-loops are an archaism or intended.

Reply:
> Do not replace looks over members with foreach. The reason is because many routines, as part of their work, add members to `members` and foreach assumes that the loop length does not change.

Maybe we should add an `opApply` to `Array` which does the right thing by default?

Reply:
An array allows arbitrary insertion and removal, so there's no single right way to iterate over an array while modifying it during iteration.

Reply:
True, but those situations would require an index-based loop anyways. `opApply` would be an improvement over the current `opSlice` based iteration  because it does the right thing in the most common cases (no change or appending).